addons/l10n_in/models/company.py

```diff
@@ -174,6 +175,31 @@ def _update_l10n_in_is_gst_registered(self):
             if company.country_code == "IN" and company.vat:
                 company.l10n_in_is_gst_registered = company.partner_id.check_vat_in(company.vat)
 
+    def _load_l10n_in_branch_journals(self):
+        """Load journals for Indian branches having different GSTIN than parent company."""
+
+        # Consider only Indian branches with GST and a parent company
+        for branch in self.filtered(
+            lambda b: b.parent_id and b.vat and b.country_code == 'IN'
```

Comment:
```suggestion
            lambda b: b.vat and b.country_code == 'IN'
```
There will always be a parent_id, a company is a parent of itself